source/PlayerInfoPanel.cpp

```diff
@@ -372,45 +466,57 @@ bool PlayerInfoPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &comman
 
 bool PlayerInfoPanel::Click(int /* x */, int /* y */, int clicks)
 {
+	for(auto &zone : menuZones)
+		if(zone.Contains(UI::GetMouse()))
+		{
+			SortShips(*zone.Value());
+			panelState.SetCurrentSort(zone.Value());
+			return true;
+		}
+	
 	// Do nothing if the click was not on one of the ships in the fleet list.
 	if(hoverIndex < 0)
 		return true;
 	
 	bool shift = (SDL_GetModState() & KMOD_SHIFT);
 	bool control = (SDL_GetModState() & (KMOD_CTRL | KMOD_GUI));
-	if(canEdit && (shift || control || clicks < 2))
+	if(panelState.CanEdit() && (shift || control || clicks < 2))
 	{
 		// Only allow changing your flagship when landed.
-		if(control && allSelected.count(hoverIndex))
-			allSelected.erase(hoverIndex);
+		if(control && panelState.AllSelected().count(hoverIndex))
+			panelState.AllSelected().erase(hoverIndex);
 		else
 		{
-			if(allSelected.count(hoverIndex))
+			if(panelState.AllSelected().count(hoverIndex))
 			{
 				// If the click is on an already selected line, start dragging
 				// but do not change the selection.
 			}
 			else if(control)
-				allSelected.insert(hoverIndex);
+				panelState.AllSelected().insert(hoverIndex);
 			else if(shift)
 			{
 				// Select all the ships between the previous selection and this one.
-				for(int i = max(0, min(selectedIndex, hoverIndex)); i <= max(selectedIndex, hoverIndex); ++i)
-					allSelected.insert(i);
+				int start = max(0, min(panelState.SelectedIndex(), hoverIndex));
+				int end = max(panelState.SelectedIndex(), hoverIndex);
+				for(int i = start; i <= end; ++i)
+					panelState.AllSelected().insert(i);
 			}
 			else
 			{
-				allSelected.clear();
-				allSelected.insert(hoverIndex);
+				panelState.AllSelected().clear();
+				panelState.AllSelected().insert(hoverIndex);
 			}
-			selectedIndex = hoverIndex;
+			panelState.SetSelectedIndex(hoverIndex);
 		}
 	}
 	else
 	{
 		// If not landed, clicking a ship name takes you straight to its info.
+		panelState.SetSelectedIndex(hoverIndex);
+		
 		GetUI()->Pop(this);
-		GetUI()->Push(new ShipInfoPanel(player, hoverIndex));
+		GetUI()->Push(new ShipInfoPanel(player, move(panelState)));
```

Comment:
Prefer `std::move` over an unqualified `move` (this is not strictly necessary but is a way of highlighting important and uncommon calls. Moving an object means we need to take special care with the moved-from object.
```suggestion
		GetUI()->Push(new ShipInfoPanel(player, std::move(panelState)));
```